tesseract_collision/include/tesseract_collision/contact_checker_common.h

```diff
@@ -124,6 +129,233 @@ inline ContactResult* processResult(ContactDistanceData& cdata,
+  {
+    btVector3& v = conv.vertices[i];
+    vertices.push_back(Eigen::Vector3d(v.getX(), v.getY(), v.getZ()));
+  }
+
+  int num_faces = conv.faces.size();
+  faces.reserve(3 * num_faces);
+  for (int i = 0; i < num_faces; i++)
+  {
+    std::vector<int> face;
+    face.reserve(3);
+
+    const btConvexHullComputer::Edge* sourceEdge = &(conv.edges[conv.faces[i]]);
+    int a = sourceEdge->getSourceVertex();
+    face.push_back(a);
+
+    int b = sourceEdge->getTargetVertex();
+    face.push_back(b);
+
+    const btConvexHullComputer::Edge* edge = sourceEdge->getNextEdgeOfFace();
+    int c = edge->getTargetVertex();
+    face.push_back(c);
+
+    edge = edge->getNextEdgeOfFace();
+    c = edge->getTargetVertex();
+    while (c != a)
+    {
+      face.push_back(c);
+
+      edge = edge->getNextEdgeOfFace();
+      c = edge->getTargetVertex();
+    }
+    faces.push_back(face.size());
+    faces.insert(faces.end(), face.begin(), face.end());
+  }
+
+  return num_faces;
+}
+
+/**
+ * @brief Write a simple ply file given vertices and faces
+ * @param path The file path
+ * @param vertices A vector of vertices
+ * @param faces The first values indicates the number of vertices that define the face followed by the vertice index
+ * @param num_faces The number of faces
+ */
+inline void writeSimplePlyFile(const std::string& path, const VectorVector3d vertices, const std::vector<int> &faces, int num_faces)
+{
+//  ply
+//  format ascii 1.0           { ascii/binary, format version number }
+//  comment made by Greg Turk  { comments keyword specified, like all lines }
+//  comment this file is a cube
+//  element vertex 8           { define "vertex" element, 8 of them in file }
+//  property float x           { vertex contains float "x" coordinate }
+//  property float y           { y coordinate is also a vertex property }
+//  property float z           { z coordinate, too }
+//  element face 6             { there are 6 "face" elements in the file }
+//  property list uchar int vertex_index { "vertex_indices" is a list of ints }
+//  end_header                 { delimits the end of the header }
+//  0 0 0                      { start of vertex list }
+//  0 0 1
+//  0 1 1
+//  0 1 0
+//  1 0 0
+//  1 0 1
+//  1 1 1
+//  1 1 0
+//  4 0 1 2 3                  { start of face list }
+//  4 7 6 5 4
+//  4 0 4 5 1
+//  4 1 5 6 2
+//  4 2 6 7 3
+//  4 3 7 4 0
+  std::ofstream myfile;
+  myfile.open (path);
+  myfile << "ply\n";
+  myfile << "format ascii 1.0\n";
+  myfile << "comment made by tesseract\n";
+  myfile << "element vertex " << vertices.size() << "\n";
+  myfile << "property float x\n";
+  myfile << "property float y\n";
+  myfile << "property float z\n";
+  myfile << "element face " << num_faces << "\n";
+  myfile << "property list uchar uint vertex_indices\n";
+  myfile << "end_header\n";
+
+  // Add vertices
+  for (const auto& v : vertices)
+  {
+    myfile << std::fixed << v[0] << " " << v[1] << " " << v[2] << "\n";
```

Comment:
Is there any concern about precision here? E.g. [this link](https://en.cppreference.com/w/cpp/io/manip/setprecision)

Reply:
Could be should I just set it to max precision?

Reply:
I don't know if there are any special considerations for PLY files, but you probably can't go wrong with the max precision from the example:
```c++
std::setprecision(std::numeric_limits<T>::digits10 + 1)
```